### DISCUSSION TITLE
Bep20 contract support

### DISCUSSION THREAD

**Discussion by RajKumar7777** [trustwallet/wallet-core#1571] [Q&A]
Hi,

   How to interact with BSC bep20 token using dynamic smart contract method.Please suggest any example.

Thank You

**Comment by optout21** [comment]
Please look for Ethereum examples, Binance Smart Chain uses Ethereum smart contracts.

**Comment by RajKumar7777** [comment]
@catenocrypt where can I found the Ethereum examples, please refer the link

**Reply by optout21** [reply]
I'm not sure what are you looking for exactly, best if you do search in wallet-core repo yourself.
Here are some examples:

https://github.com/trustwallet/wallet-core/blob/master/swift/Tests/Blockchains/EthereumTests.swift
https://github.com/trustwallet/wallet-core/blob/master/tests/Ethereum/ContractCallTests.cpp
https://github.com/trustwallet/wallet-core/blob/master/tests/Ethereum/TWAnySignerTests.cpp#L122

**Comment by RajKumar7777** [comment]
Hi,
 
   https://github.com/skywinder/web3swift/blob/master/Documentation/Usage.md#send-ether
   Following this link there is a method that is: Write Transaction and call smart contract method.
   By this method I can interact with contract method using contract ABI and I can do transaction for ETH and ERC20 Token.

   Like above I want to do for BSC and BEP20 Token.How to do that please support me.

**Reply by RajKumar7777** [reply]
@catenocrypt any support??

**Reply by jiuyaopinyin** [reply]
Is your test done now please? I have the same problem

**Comment by TheHmmka** [comment]
Any progress? BSC and BEP20 are still the mystic... =(
